src/Application.vala

```diff
@@ -172,14 +174,43 @@ public class Terminal.Application : Gtk.Application {
             return 0;
         }
 
+        /* If working directory not specified get from any command supplied with a full path */
```

Comment:
Is there a reason this isn't inside the conditional below? It seems like if `option_help` or `option_version` are true, then this wouldn't be relevant would it?

Reply:
I usually take a comment to refer to the code following it but in this case it could be inside the conditional - I'll move it.

Reply:
Sorry I meant the whole code block, not just the comment. It seems like we don't need to get the current working directory if the user is just printing the help or version right?

Reply:
Oh, I see. Well, it would make the following code block longer and more difficult to follow.  It is clearer (imo) to make sure that the working directory is always set to something as a fallback before processing the flags (even though in some cases the fallback is not used).  The whole thing could be pasted after the second `else` (line 202) but all that does is add an extra indent. Another alternative is to repeat the test for null working directory four times.

Reply:
I'll rethink it.